src/test/unit_tests/jest_tests/test_config_dir_restructure_upgrade_script.test.js

```diff
@@ -0,0 +1,654 @@
+        await assert_name_index_is_updated(new_account_data);
+    });
+
+    it('create_account_name_index_if_missing() - new name index already exists', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data, { symlink_access_key: false });
+        await write_manual_config_file(CONFIG_TYPES.ACCOUNT, default_config_fs, new_account_data, undefined,
+            { symlink_access_key: false, symlink_name: true });
+        const identity_path = default_config_fs.get_identity_path_by_id(new_account_data._id);
+        const account_upgrade_params = { ...new_account_data, account_name: new_account_data.name, identity_path };
+        await create_account_name_index_if_missing(default_config_fs, account_upgrade_params, dbg);
+        await assert_name_index_is_updated(new_account_data);
+    });
+
+    it('create_account_name_index_if_missing() - new name index already exists and points to a non existing location', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data, { symlink_access_key: false });
+        await write_manual_config_file(CONFIG_TYPES.ACCOUNT, default_config_fs, new_account_data, undefined,
+            { symlink_access_key: false, symlink_name: false });
+        const mock_link_id_path = path.join(default_config_fs.identities_dir_path, 'mock_id_dir', 'identity.json');
+        await symlink_account_name(default_config_fs, new_account_data.name, mock_link_id_path);
+        const identity_path = default_config_fs.get_identity_path_by_id(new_account_data._id);
+        const account_upgrade_params = { ...new_account_data, account_name: new_account_data.name, identity_path };
+        await create_account_name_index_if_missing(default_config_fs, account_upgrade_params, dbg);
+        await assert_name_index_is_updated(new_account_data);
+    });
+
+    it('create_account_name_index_if_missing123 - new name index already exists and points to wrong location', async () => {
+        const new_account_data = { _id: String(1), name: 'old_account' + 1, user: 'root', access_keys: [{ access_key: mock_access_key }] };
+        await write_manual_old_account_config_file(default_config_fs, new_account_data, { symlink_access_key: false });
+        await write_manual_config_file(CONFIG_TYPES.ACCOUNT, default_config_fs, new_account_data, undefined,
+            { symlink_access_key: false, symlink_name: false });
+        const mock_id_dir = path.join(default_config_fs.identities_dir_path, 'mock_id_dir');
+        const mock_link_id_path = path.join(mock_id_dir, 'identity.json');
+        await create_fresh_path(mock_id_dir);
```

Comment:
Could you switch the lines? (To see the `create_fresh_path` right after `mock_id_dir` was defined)